src/bootman/bootman.c

```diff
@@ -446,6 +457,210 @@ bool boot_manager_set_uname(BootManager *self, const char *uname)
         return self->have_sys_kernel;
 }
 
+static bool boot_manager_is_initrd_valid_name(const char *name)
+{
+        if (strlen(name) < 3) {
+                return false;
+        }
+
+        return isdigit(name[0]) && isdigit(name[1]);
+}
+
+/**
+ * initrd name must start with two numbers in order to prioritize execution
+ * Note: at this moment systemd-boot not support muli inird usage
+ */
+static int boot_manager_sort_initrd_name(const void *name1, const void *name2)
+{
+        return strcmp((char *)name1, (char *)name2);
+}
+
+bool boot_manager_get_initrd_freestanding(BootManager *self)
+{
+        autofree(DIR) *initrd_dir = NULL;
+        NcArray *dir_files = NULL;
+        struct dirent *ent = NULL;
+        struct stat st = { 0 };
+
+        if (!self || !self->initrd_freestanding_dir) {
+                return false;
+        }
+
+        dir_files = nc_array_new();
+        OOM_CHECK(dir_files);
+
+        initrd_dir = opendir(self->initrd_freestanding_dir);
+        if (!initrd_dir) {
+                LOG_ERROR("Error opening %s: %s", self->initrd_freestanding_dir, strerror(errno));
+                nc_array_free(&dir_files, NULL);
+                return false;
```

Comment:
So I'm not sure this should be an error. If there isn't a distro initrd directory I think that we should just continue and probably make the log info level.

Reply:
the error is for try to open this folder if there is not initrd the folder must not exist?

Reply:
Right, if the initrd folder doesn't exist I don't think that is fatal, it just means there is no free standing initrd.

Reply:
Maybe that makes some of the validation you are doing later on harder though, I'm not sure.

Reply:
changing it